ovh/types_iam.go

```diff
@@ -0,0 +1,103 @@
+package ovh
+
+type IamReferenceAction struct {
+	Action       string   `json:"action"`
+	Categories   []string `json:"categories"`
+	Description  string   `json:"description"`
+	ResourceType string   `json:"resourceType"`
+}
+
+func (a *IamReferenceAction) ToMap() map[string]any {
+	out := make(map[string]any, 4)
+
+	out["action"] = a.Action
+	out["categories"] = a.Categories
+	out["description"] = a.Description
+	out["resource_type"] = a.ResourceType
+
+	return out
+}
+
+type IamPolicy struct {
+	Id          string         `json:"id,omitempty"`
+	Name        string         `json:"name"`
+	Description string         `json:"description,omitempty"`
+	Identities  []string       `json:"identities"`
+	Resources   []IamResource  `json:"resources"`
+	Permissions IamPermissions `json:"permissions"`
+	CreatedAt   string         `json:"createdAt,omitempty"`
+	UpdatedAt   string         `json:"updatedAt,omitempty"`
+	ReadOnly    bool           `json:"readOnly,omitempty"`
+	Owner       string         `json:"owner,omitempty"`
+}
+
+func (p IamPolicy) ToMap() map[string]any {
+	out := make(map[string]any, 0)
+	out["name"] = p.Name
+	if p.Description != "" {
+		out["description"] = p.Description
+	}
+
+	out["identities"] = p.Identities
+	out["owner"] = p.Owner
+	out["read_only"] = p.ReadOnly
+	out["created_at"] = p.CreatedAt
+	if p.UpdatedAt != "" {
+		out["updated_at"] = p.UpdatedAt
+	}
+	var resources []string
+	for _, r := range p.Resources {
+		resources = append(resources, r.URN)
+	}
+	out["resources"] = resources
+
+	// inline allow and except
+	allow, except := p.Permissions.ToLists()
+	out["allow"] = allow
+	out["except"] = except
```

Comment:
Allow and except can be empty, if it is the case, should I skip setting them or should I set them to the empty list like I am doing now ?